nemo/src/static_checks/rule_set.rs

```diff
@@ -0,0 +1,571 @@
+//! Functionality that provides methods in relation with Rule-Types.
+use crate::rule_model::components::{
+    atom::Atom, rule::Rule, term::primitive::variable::Variable, IterableVariables,
+};
+use crate::static_checks::collection_traits::{Disjoint, InsertAll, Superset};
+use crate::static_checks::positions::{Position, Positions, PositionsByRuleAndVariables};
+
+use std::collections::{HashMap, HashSet};
+
+/// Type to relate an (existential) Variable to a Rule. Therefore a type to identificate an
+/// (existential) Variable.
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub struct RuleAndVariable<'a>(pub &'a Rule, pub &'a Variable);
+
+impl Ord for RuleAndVariable<'_> {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        self.1.cmp(other.1)
+    }
+}
+
+impl PartialOrd for RuleAndVariable<'_> {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.1.cmp(other.1))
+    }
+}
+
+// TODO: USE A REFERENCE TO RULE
+/// Type to wrap a 'set' (Vec) of Rule(s).
+#[derive(Debug)]
+pub struct RuleSet(pub Vec<Rule>);
+
+/// Type to relate 2 Variable(s) together.
+#[derive(Debug, Eq, Hash, PartialEq)]
+pub struct RuleAndVariablePair<'a>(pub [RuleAndVariable<'a>; 2]);
+
+// NOTE: KEEP?
+/// This Trait provides methods to get all (HashSet<Position> / Positions) of some type.
+pub trait AllPositivePositions<'a> {
```

Comment:
Just because a method is implemented on two different structures does not mean that there needs to be a trait for this. 
For this trait and also others in this file I would say that the functionality is too specific to justify generalizing this into a trait.
If you want to keep the trait, it needs to describe a proper capability.

Also `PositivePosition` should be renamed to something like `PositiveLiteralPosition` because it is not clear what it means for a `Position` to be positive.